sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/ObjectHashAggregateSuite.scala

```diff
@@ -457,4 +458,38 @@ class ObjectHashAggregateSuite
       )
     }
   }
+
+  test("SPARK-46536 Support GROUP BY CalendarIntervalType") {
+    withSQLConf(
+      SQLConf.USE_OBJECT_HASH_AGG.key -> "true",
+      SQLConf.OBJECT_AGG_SORT_BASED_FALLBACK_THRESHOLD.key -> "1"
+    ) {
+
+      val numRows = 50
+
+      val dfSame = createAggregate(_ => Tuple1(new CalendarInterval(1, 2, 3)), numRows)
+      assert(getPlanOutput(dfSame).contains("HashAggregate"))
```

Comment:
there are two hash aggregate operators in Spark: `HashAggregateExec` and `ObjectHashAggregateExec`. Can we test both?